grails-app/controllers/org/pih/warehouse/user/DashboardController.groovy

```diff
@@ -149,13 +149,24 @@ class DashboardController {
         render results as JSON
     }
 
-    @Cacheable("megamenuCache")
     def megamenu = {
+        Location location = Location.get(session.warehouse?.id)
+        if (!location.supports(ActivityCode.MANAGE_INVENTORY) && location.supports(ActivityCode.SUBMIT_REQUEST)) {
+            return [ menu: [] ]
+        }
+
+        Map menuConfig = grailsApplication.config.openboxes.megamenu;
+        menuConfig = menuConfig.findAll { key, val -> key != "configuration" }
+        User user = User.get(session?.user?.id)
+
+        if (userService.hasHighestRole(user, session?.warehouse?.id, RoleType.ROLE_AUTHENTICATED)) {
+            menuConfig = grailsApplication.config.openboxes.requestorMegamenu;
+        }
+        List translatedMenu = megamenuService.buildAndTranslateMenu(menuConfig, user, location).findAll { !it.isEmpty() }
```

Comment:
Can you explain the filter here? And maybe add a comment.

Reply:
There was a case where if menu section didn't have following attributes: `href`, `subsections`, `menuItems` it would return an empty object `return [:]` 
In my latest commit I figured that instead of filtering them out I should probably prevent the empty objects from being added to the megamenu list

**MegamenuService.groovy >  buildAndTranslateSections**
```
 def translatedSections = buildAndTranslateSections(value, key, user, location)
 if (translatedSections) {
    parsedMenuConfig << translatedSections
}
```